azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -254,10 +242,8 @@ public Project getProject(final String name) {
    * fetch active project from cache and inactive projects from db by project_id
    */
   public Project getProject(final int id) {
-    Project fetchedProject = null;
-    if (isActiveProject(id)) {
-      fetchedProject = this.projectsById.get(id);
-    } else {
+    Project fetchedProject = this.projectsById.get(id);
```

Comment:
This is a clever change. 

I assume this is designed to address your goal
> When one user is removing the project while the other user is fetching the project, the correct project will be returned instead of null.

Reply:
Yes, I combined "containsKey" and "get" operations together so that we can rely on the concurrent map itself for thread safety without adding additional external synchronization.